src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -318,18 +329,32 @@ def trigger_aggregator_commands(self, command):
         try:
             if command["type"] == "bid":
                 return self._bid_aggregator(command)
+            elif command["type"] == "update_bid":
+                return self._update_bid_aggregator(command)
             elif command["type"] == "delete_bid":
                 return self._delete_bid_aggregator(command)
             elif command["type"] == "list_bids":
                 return self._list_bids_aggregator(command)
             elif command["type"] == "offer":
                 return self._offer_aggregator(command)
+            elif command["type"] == "update_offer":
+                return self._update_offer_aggregator(command)
             elif command["type"] == "delete_offer":
                 return self._delete_offer_aggregator(command)
             elif command["type"] == "list_offers":
                 return self._list_offers_aggregator(command)
             elif command["type"] == "device_info":
                 return self._device_info_aggregator(command)
+            elif command["type"] == "list_market_stats":
+                market_data = self.device.parent.stats.get_market_stats(
+                    command['data']['market_slots']
+                )
+                return {
+                    "command": "list_market_stats", "status": "ready",
+                    "market_data": market_data,
+                    "transaction_id": command.get("transaction_id", None),
+                    "area_uuid": self.device.uuid
+                }
```

Comment:
Please refactor this into its own function, just like the other commands. Thanks!